airbyte-integrations/connectors/source-stripe/source_stripe/source.py

```diff
@@ -104,11 +107,87 @@ def validate_and_fill_with_defaults(config: MutableMapping[str, Any]) -> Mutable
+        """
+        Check if the source connector uses OAuth2 authentication.
+
+        Args:
+            config: source connector configuration.
+        """
+        if "credentials" in config:
+            return config["credentials"]["credentials_title"] == "OAuth Credentials"
+
+        return False
+
+    def has_client_secret_auth(self, config: Mapping[str, Any]) -> bool:
+        """
+        Check if the source connector uses Client Secret authentication.
+
+        Args:
+            config: source connector configuration.
+        """
+        if "credentials" in config:
+            return config["credentials"]["credentials_title"] == "Client Secret Credentials"
+
+        return False
+
+    def set_authenticator(self, config: MutableMapping[str, Any]):
+        """
+        Set the authenticator for the source connector.
+
+        Args:
+            config: source connector configuration.
+        """
+        credentials = config["credentials"]
+        credentials_title = credentials["credentials_title"]
+
+        if self.has_oauth2_auth(config):
+            self._authenticator = StripeOauth2Authenticator(
+                token_refresh_endpoint="https://connect.stripe.com/oauth/token",
+                client_id=credentials["client_id"],
+                client_secret=credentials["client_secret"],
+                refresh_token=credentials["refresh_token"],
+            )
+        elif self.has_client_secret_auth(config):
+            self._authenticator = TokenAuthenticator(credentials["client_secret"])
+        else:
+            raise ValueError(f"Unknown credentials_title: {credentials_title}")
+
+    def get_authenticator(self, config: Mapping[str, Any]) -> StripeOauth2Authenticator | TokenAuthenticator:
+        """
+        Get the authenticator for the source connector.
+
+        Args:
+            config: source connector configuration.
+        """
+        if self._authenticator is None:
+            self.set_authenticator(config)
+
+        return self._authenticator
+
+    def get_access_token(self, config: Mapping[str, Any]) -> str:
+        """
+        Get the access token for the source connector.
+
+        Args:
+            config: source connector configuration.
+        """
+        authenticator = self.get_authenticator(config)
+
+        if self.has_oauth2_auth(config):
+            return authenticator.get_access_token()
+        elif self.has_client_secret_auth(config):
+            return authenticator._token
+        else:
+            raise ValueError("Unknown authentication")
+
     def check_connection(self, logger: logging.Logger, config: MutableMapping[str, Any]) -> Tuple[bool, Any]:
-        args = self._get_stream_base_args(config)
-        account_stream = StripeStream(name="accounts", path="accounts", use_cache=USE_CACHE, **args)
         try:
-            next(account_stream.read_records(sync_mode=SyncMode.full_refresh), None)
```

Comment:
I am replacing this new code implemented a couple of days ago because it does not work with tokens that do not have full access to connected accounts.